### PR TITLE
Fix OSGeo hook for versions 2.0+

### PR DESCRIPTION
**Pull request by davidh-ssec**
See #1522.

Not sure this is the best way to solve this, but it should work. It should really be tested with different versions and platforms.

I should note that I made these changes based on the previous loader build because the newest one doesn't load on my system (glibc problems, see #2191). So technically I haven't tested these changes with the bootloader in this PR, but it should work.

### CODE REVIEW

File: PyInstaller/loader/rthooks/pyi_rth_osgeo.py [hunk 1]
```diff
@@ -20,3 +20,7 @@
 
 if os.path.exists(gdal_data):
     os.environ['GDAL_DATA'] = gdal_data
+
+# We *must* import site so that the 'help' builtin is added for Anaconda
```
Comment (giumas) [inline PyInstaller/loader/rthooks/pyi_rth_osgeo.py]: Are you sure that this is only required for Anaconda?

Comment (davidh-ssec) [inline PyInstaller/loader/rthooks/pyi_rth_osgeo.py]: It is the only environment that I saw in the issue of having this problem, but based on what you've shown it does look like this must be for everything.

File: PyInstaller/fake-modules/site.py [hunk 2]
```diff
@@ -56,3 +56,11 @@ def __call__(self, *args, **kwds):
...
          pydoc = __import__(''.join('pydoc'))
          return pydoc.help(*args, **kwds)
+
+try:
+    import __builtin__ as builtins # Python 2
+except ImportError:
+    import builtins # Python 3
+# OSGeo wraps the builtin help function even though it is not available
+# https://github.com/pyinstaller/pyinstaller/issues/1522
+builtins.help = _Helper()
```
Comment (davidh-ssec) [inline PyInstaller/fake-modules/site.py]: Hm newest problem, how do I check the osgeo version while the rt hook is running?

Comment (davidh-ssec) [inline PyInstaller/fake-modules/site.py]: I'm starting to think the best/cleanest option it to just not support gdal versions affected by this.

**Comment by htgoebel** [comment]
@davidh-ssec Any news on this pull-request?

**Comment by davidh-ssec** [comment]
No, I think I should remove it based on @giumas comments. I'll go ahead and close it.

**Pull request closed without merging**